### DISCUSSION TITLE
'Install' button gives misleading instructions. Make it link to "Installation" in the readme instead

### DISCUSSION THREAD

**Discussion by haesleinhuepf** [chanzuckerberg/napari-hub#350] [Ideas]
#### Description

Hi all,

potential new napari-plugin users might read misleading instructions when clicking the `Install` button on the napari hub page of a plugin. If those instructions don't work, that's unfortunate especially for users who try for the first time. 

These plugins are affected for sure to my knowledge, at least on Windows, because you need to use conda to install prerequesites:
* [napari-accelerated-pixel-and-object-classification](https://www.napari-hub.org/plugins/napari-accelerated-pixel-and-object-classification#installation)
* [napari-pyclesperanto-assistant](https://www.napari-hub.org/plugins/napari-pyclesperanto-assistant#installation)
* [napari-script-editor](https://www.napari-hub.org/plugins/napari-script-editor#installation)
* [napari-clusters-plotter](https://www.napari-hub.org/plugins/napari-clusters-plotter#installation)

Furthermore, I see also related discussions here:
* https://github.com/napari/napari/issues/3223
* https://github.com/stardist/stardist-napari/issues/3
* https://github.com/napari/napari/issues/3631
* https://forum.image.sc/t/error-when-installing-napari-omero-through-napari-bundled-app/60171/4
* https://forum.image.sc/t/best-practice-for-scripting-napari-for-end-users/60077/20
* https://forum.image.sc/t/can-not-install-napari-stardist/60085
* https://github.com/chanzuckerberg/napari-hub/discussions/334
* https://github.com/chanzuckerberg/napari-hub/discussions/331

#### Steps/Code to Reproduce

On a Windows computer, in a napari bundle app, try to install one of the plugins listed above by following the instructions after clicking the `Install` button on the napari hub.

#### Expected Results

It would be great if clicking the `Install` button would lead to the Installation instructions written by the developers and not to generic instructions that don't work.

Potential alternative: Make the link behind the button configurable and keep its current behaviour as default in case no link is configured.

Thanks!

Best,
Robert

**Comment by DragaDoncila** [comment]
> Potential alternative: Make the link behind the button configurable and keep its current behaviour as default in case no link is configured.

Since many plugin developers may not have an Installation section, I like this option. We could have this as one of the project links in `config.yaml` and defer to the `pip install` instructions if this is not explicitly configured by developers.

**Comment by neuromusic** [comment]
> We could have this as one of the project links in config.yaml and defer to the pip install instructions if this is not explicitly configured by developers.

or `.napari/INSTALL.md` to replace the copy in the modal?

**Comment by neuromusic** [comment]
Thanks for bringing this up @haesleinhuepf !!

When we launched and added the "Install" button, our hope was that we would be able to soon replace the text in the popup with #242, however, as the plugin ecosystem has grown and these issues have surfaced, its clear that the simple instructions are insufficient for many plugins.

I'm optimistic that we can get closer to simplicity for users in 2022 with the work happening on the bundled app with conda support, but in the meantime, I agree that offering support for more complex/advanced installation instructions will be important.

**Comment by DragaDoncila** [comment]
> or .napari/INSTALL.md to replace the copy in the modal?

This is probably an easier way to provide the content for the modal, but I do worry about the developers having to maintain so many files e.g. `README`, `DESCRIPTION` and now `INSTALL` which they might want in the other two documents? I suppose we could suggest that they `include` the `INSTALL` doc if they want it in multiple places?

**Comment by haesleinhuepf** [comment]
I like the way you both approach this issue. And I agree, the simpler, the better, for both users and developers.

Btw. almost all plugins come with an "Installation" section, because it's part of the plugin template. Thus, I bet this link works in 95% of the cases independent from which [plugin] you enter:

https://napari-hub.org/plugins/[plugin]#installation

**Comment by neuromusic** [comment]
> I do worry about the developers having to maintain so many files

agreed, but I think the problem is redundancy, not the number of files _per se_. in other words, plugin developers should only have to specify installation instructions once and this should be easily accessed from Github (via a shield?) or the napari hub page.

**Comment by neuromusic** [comment]
as the "stardist on windows" blog post highlights, much of the complexity here comes down to tailoring the installation to the user's environment (Win vs Mac vs Linux, for example) OR configuration options (such as [cellpose's different installation paths for CPU and GPU](https://www.napari-hub.org/plugins/cellpose-napari#installation)).

**Comment by neuromusic** [comment]
Since the current functionality is consistent with how we designed the "Install button" feature (e.g. its not immediately actionable as a bug, but needs some design and implementation considerations), I'm going to move this to Discussions so we can try to tackle a better solution in Q1
